_posts/2020-04-03-why-did-we-switch-to-quarkus.adoc

```diff
@@ -0,0 +1,85 @@
+
+> A Kubernetes native Java stack tailored for OpenJDK HotSpot and GraalVM, crafted from the best of breed Java libraries and standards.
+
+On Quarkus website you can find information that it has been designed around "container first" philosophy, meaning that it is optimized for fast boot times and low memory usage. As a result, we should gain near instant scale-up and high-density memory utilization in container orchestration platforms.
+
+The way this has been achieved can be summarized as follows:
+
+* *Build time metadata processing.* As much processing as possible is squeezed into the build process e.g. all of the classes required to perform initial application deployment are used and dropped while building. They are not loaded into final JARs (and JVMs) or native executables. You can say that the framework is tailor-made for you every time you press a build button, without compromising the effectiveness of a declarative approach.
+
+* *Reduction in reflection usage.* Considering the point above, the resulting application contains only direct invocations. All of the fields, methods and classes that are not being called directly are purposefully removed. The same applies to dependencies as well. There is a way to include such calls (register them for reflection), but Quarkus will gently warn you that you may be deflecting from the real purpose of the framework.
+
+* *Native image pre-boot.* Quarkus supports native executables that can be run with GraalVM. Actually, creators of Quarkus state that the framework was designed to be run as a native image. The ahead-of-time compilation reduces the need of dynamic scanning and loading classes into a JVM.
+
+* *GraalVM.* GraalVM is a native code generator which will compile your project into an executable that can be run without the JVM. You can read more about this superb project here: https://www.graalvm.org/.
+
+Below you can see a comparison of memory usage and boot time of sample applications:
+
+image::perf_black.png[Performance Chart]
+source: https://quarkus.io/
+
+== Evolution, not revolution
+
+What is interesting about this framework is that it is not revolutionary. Since some of Payair’s developers had mainly Spring experience, we were concerned that it would be difficult for them to “switch sides”. It turned out that our fear of the unknown was completely unfounded. Quarkus is built on the foundations of good old Jakarta EE standards that all Java developers are familiar with. We did not have to learn a bunch of new APIs that are advertised as "the best way to create REST endpoint" or "reinvented database access". Instead, Quarkus uses Context Dependency Injection for beans, JAX-RS for RESTful web services, JPA for database access and many others.
+
+If you happen to write your project using Jakarta EE, switching to Quarkus would be as easy as changing the runtime processor. Majority of your old code would work without any adjustments.
```

Comment:
Suggest the following modification:
"If you are a Jakarta EE or Spring Boot developer, switching to Quarkus would be as easy. Picking up Quarkus for us was almost effortless."

Reply:
I'd like to point out here that switching to Quarkus is especially easy for Jakarta EE developers. If it doesn't fit you, we can simply remove this paragraph.

Reply:
I was softening your statement because we don't want to give the impression that if you just switch your runtime to Quarkus for your Jakarta EE application, everything will work just fine. Considering running existing Jakarta EE applications on Quarkus need to be carefully considered on a case-by-case basis.
I'd suggest removing the paragraph then please. Thank you